Sources/MuxUploadSDK/Upload/ChunkedFileUploader.swift

```diff
@@ -88,15 +90,28 @@ class ChunkedFileUploader {
             do {
                 // It's fine if it's already open, that's handled by ignoring the call
                 try file.openFile(fileURL: uploadInfo.videoFile)
+                let fileSize = file.fileSize
                 try file.seekTo(byte: lastByte)
                 let result = try await makeWorker().performUpload()
                 file.close()
-                
+
                 let success = UploadResult(
                     finalProgress: result.progress,
                     startTime: result.startTime,
                     finishTime: result.updateTime
                 )
+
+                let asset = AVAsset(url: uploadInfo.videoFile)
+
+                var duration: CMTime
+                if #available(iOS 16, *) {
+                    duration = try await asset.load(.duration)
+                } else {
+                    duration = asset.duration
+                }
```

Comment:
I worry that this could cause the SDK to hang while it calculates the duration. Is there a better way to do this or is it ok?